### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by felix-ht** [m0nac0/flutter-mapbox-gl#295]
What are the chances that we could implement Desktop/Windows support for this package. https://github.com/maplibre/maplibre-native Seems to have a build option for windows. But the binding code seems to be missing/lacking.

What would it take to get a plugin going for flutter? I would be willing to pay a bounty for this task.

**Comment by wipfli** [comment]
Hi Felix. Thanks for the suggestion. Not sure, but maybe I can put you in touch with some people who might be interested in working on this. Can you text me in the MapLibre slack? My username there is `@wipfli`

**Comment by m0nac0** [comment]
I think we would first need support for platform-views on desktop platforms in Flutter. 
I have been tracking this for quite some time, but I don't know how close the Flutter team is to implementing this.
It is on their roadmap for 2023: https://github.com/flutter/flutter/wiki/Roadmap#features

For Windows specifically the Flutter issue is https://github.com/flutter/flutter/issues/31713
There were PRs in the past for macOS and Linux, but if I remember correctly they were abandoned for the time being to rethink the API design or something like that. You should be able to find them with the GitHub search in the flutter repo.

**Comment by felix-ht** [comment]
thanks for the details @m0nac0 This certainly is a blocker for before any work can be done here.

@wipfli thanks for the proposal - but as we rely on flutter we probably will have to wait until something moves on that side

**Comment by m0nac0** [comment]
There now seems to be a design doc for Windows platform views: https://docs.google.com/document/d/10nVF2f-QTQ5o2HMEZ1fUlBJzK4dEmhuPHTVFEgW-JRs/edit?resourcekey=0-J-Lsp7xfA-Nr5EMr1av-Xg

**Comment by josxha** [comment]
Open issue here
- https://github.com/flutter/flutter/issues/31713

**Comment by rekire** [comment]
It is crazy how many features are blocked by the PlatformView support in Flutter. Even worse that this is still the case since 2019

**Comment by josxha** [comment]
@rekire not necessarily. There would be an alternative implementation by using the [Texture API](https://api.flutter.dev/flutter/widgets/Texture-class.html). There is even an [early prototype](https://github.com/kekland/maplibre_flutter_map) for the `flutter_map` package some time ago. Bindings to the MapLibre Core library could be used from every plattform except maybe web. I think @sargunv used an equivalent when he implemented desktop support for maplibre-compose in https://github.com/maplibre/maplibre-compose/pull/544 .
Platform Views would just be the easiest implementation, but given that it is labeled `P2` and nobody is working on it, we probably won't see it any time soon.

**Comment by sargunv** [comment]
I'm not familiar with flutter, but even web _should_ be possible by building the native core to WASM. I believe folks have successfully done this with MapLibre QT.
